perceval/components/linear_circuit.py

```diff
@@ -581,10 +580,10 @@ def compute_unitary(self,
     @staticmethod
     @deprecated(version="0.10.0", reason="Construct a GenericInterferometer object instead")
     def generic_interferometer(m: int,
-                               fun_gen: Callable[[int], ACircuit],
-                               shape: Union[str, InterferometerShape] = InterferometerShape.RECTANGLE,
+                               fun_gen: callable[[int], ACircuit],
```

Comment:
callable[something] is not a correct type for hinting, you must use from collections.abc import Callable.
This is not causing any error because it's not used anywhere.
You can see code example in perceval/components/generic_interferometer.py GenericInterferometer.__init__